trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/IdentifierService.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.services;
+
+import static org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.services.EndpointConstants.VERSION;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+/**
+ *
+ * Service to identify important information regarding the trace server and the system it is running on.
+ *
+ * @author Vlad Arama
+ *
+ */
+@Path("/identifier")
+@Tag(name = EndpointConstants.IDF)
+public class IdentifierService {
+    /**
+     * Getter returning important information about the system, including server version, OS, CPU count, memory size, launcher and product names.
+     *
+     * @return A JSON response containing the system's details.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Retrieves system and server information", responses = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved the system and server information")
```

Comment:
It should be documented what is mandatory and optional. I think version should be mandatory.

Reply:
I created a model class `ServerInfoResponse` with serverVersion and OS as required fields. I then created an implementation of this interface `ServerInfoResponseImpl` to use in the `IdentifierService`. I think this makes the code cleaner and better documented, but we can change the approach if there is a better way.

Reply:
The java package with all the Models are for swagger documentation only. All data structures are usually from Trace Compass mainline and we didn't want to add any dependencies to swagger or json serialization/deserilization needed for the Trace Server only. This approach is not ideal, but was a good compromise to have a manageable documentation.

The identifier service is a bit different, because it's solely in the trace server code. Still I think the model package should only contain interfaces for the documentation only.

The model package is also not exported (see MANIFEST.MF) so that downstream projects won't used it.

That would mean that ServerInfoResponseImpl can't extend the ServerInfoResponse interface.